hazelcast/src/main/java/com/hazelcast/datalink/impl/DataLinkConsistencyChecker.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.datalink.impl;
+
+import com.hazelcast.config.DataLinkConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.datalink.impl.DataLinkServiceImpl.DataLinkSource;
+import com.hazelcast.datalink.impl.DataLinkServiceImpl.DataLinkSourcePair;
+import com.hazelcast.jet.impl.JetServiceBackend;
+import com.hazelcast.map.IMap;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.QueryUtils;
+import com.hazelcast.sql.impl.schema.datalink.DataLink;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class DataLinkConsistencyChecker {
+    private final DataLinkServiceImpl dataLinkService;
+    private final IMap<Object, Object> sqlCatalog;
+
+    public DataLinkConsistencyChecker(HazelcastInstance instance, NodeEngine nodeEngine) {
+        assert nodeEngine.getDataLinkService() instanceof DataLinkServiceImpl;
```

Comment:
does this assert add any value? it has no custom message. cast in second line will always check the type anyway

Reply:
```suggestion
```

Reply:
Yeap, it was a sanity check during development. Removed.